### PR TITLE
Various docs updates

### LINKED ISSUE #236: Onboarding and developer documentation
### Definition of Done / Acceptance Criteria

Feel free to conver into sub issues and assign them, and create extra issues off this 

- [x] Top level docs directory with
  - [x] UCLH infrastructure setup
  - [x] developer setup 
  - [x] Design from the PIXL perspective
	  - https://github.com/UCLH-Foundry/the-rolling-skeleton/blob/main/docs/design/100-day-design.md has current design. Edit and commit the `.drawio` and the `.drawio.png` to an `img` or similar subdirectory so that we can update this as the design changes
	  - Think about any design decisions which aren't clear and make an issue to answer them
- [x] Top level README points to docs as appropriate 
- [x] Services READMEs have been checked and it is clear
	- [x] What their purpose is, and where they are used in the PIXL pipeline
	- [x] How to run tests
	- [x] High level functionality and entry points into the code if appropriate 
- [ ] Any documentation in terms of the wider UCLH context rather than PIXL being added as PRs to [the-rolling-skeleton](https://github.com/UCLH-Foundry/the-rolling-skeleton/)

### Testing

_No response_

### Documentation

_No response_

### Dependencies

- [x] #242
- [x] #243
- [x] #244
- [x] #245
- [x] #249 
- [x] #253  
- [x] #257 
- [x] #306 

### Details and Comments

Very rough plan but expect this will evolve, main aim is to read code and documenation together

### LINKED ISSUE #243: Document `extract-radiology-reports` CLI command
### Definition of Done / Acceptance Criteria

Brief description of the `pixl extract-radiology-reports` is added to the [PIXL CLI README](https://github.com/UCLH-Foundry/PIXL/blob/main/cli/README.md).

### Testing

_No response_

### Documentation

Update [PIXL CLI README](https://github.com/UCLH-Foundry/PIXL/blob/main/cli/README.md)

### Dependencies

_No response_

### Details and Comments

_No response_

### LINKED ISSUE #245: Create and populate `docs/` directory for PIXl
### Definition of Done / Acceptance Criteria

PIXL has a `docs/` directory containing high-level documentation and design decisions regarding the PIXL pipeline.

### Testing

_No response_

### Documentation

Add `docs/` directory with relevant markdown files.

### Dependencies

_No response_

### Details and Comments

- Do we want to move the contents of the module-level READMEs to `docs/` and link to it?
- How should we organise the `docs/` directory?
- Review contents of

### LINKED ISSUE #306: Add documentation for the PIXL database
### Definition of Done / Acceptance Criteria

- [ ] We have a `docs/services/pixl_database.md` file describing the PIXL database
- [ ] We have linked to this Markdown file from any other Markdown files that mention the PIXL database

### Testing

_No response_

### Documentation

_No response_

### Dependencies

_No response_

### Details and Comments

Part of #236.

### PR DESCRIPTION
**Pull request by milanmlft**
Fixes #236, fixes #243, fixes #245, fixes #306

### CODE REVIEW

**Comment by milanmlft** [comment]
`docs/services/pixl_database.md` could definitely do with some more fleshing out if anyone feels up for it @UCLH-Foundry/arc-dev.

File: README.md [hunk 1]
```diff
@@ -10,6 +10,21 @@ several services orchestrated by [Docker Compose](https://docs.docker.com/compos
 
 To get access to the GAE, [see the documentation on Slab](https://uclh.slab.com/posts/gae-access-7hkddxap)
 
+## Development
+
+[Follow the developer setup instructions](./docs/setup/developer.md).
+
+Before raising a PR, make sure to **run all tests** for each PIXL module
+and not just the component you have been working on as this will help us catch unintentional
+regressions without spending GitHub actions minutes :-)
```
Comment (stefpiatek) [inline README.md:19]: NGL I don't do this, as its a public project they're free 🤷🏻

File: docs/services/pixl_database.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# The PIXL database
+
+PIXL uses a [postgres database](../../postgres/README.md) to 
+
+- Add hashed identifiers along with the originals for DICOM images (in `pixl_dcmd`)
+- Keep track of the export status of images (in `core.uploader`)
```
Comment (stefpiatek) [inline docs/services/pixl_database.md:6]: I'd probably say imaging studies and the projects that they are used in

Might be worth saying that the pipeline will not process any studies for a project that have already been exported?

File: docs/setup/developer.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+There is support (sometimes through plugins) for these tools in most IDEs & editors.
+
+Before raising a PR, make sure to **run all tests** for each PIXL module
+and not just the component you have been working on as this will help us catch unintentional regressions without spending GH actions minutes :-)
+
+## Linting
+
+We run [pre-commit](https://pre-commit.com/) as part of the GitHub Actions CI. To install and run it locally, do:
+
+```sh
```
Comment (stefpiatek) [inline docs/setup/developer.md:17]: doesn't make too much of a difference but we've used `shell` elsewhere

**Review by stefpiatek — APPROVED** [review]
Nice, thanks for getting this in Milan

File: README.md (hunk 1 above)
Comment (milanmlft) [reply README.md:19]: Yeah fair, I think this is leftover from when repo was still private

**Pull request merged**